src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -356,6 +359,43 @@ private void conjureSnapshotTablesFromRowIds(SnapshotRequestRowIdModel requestRo
         }
     }
 
+    private void conjureSnapshotTablesFromLiveviews(Snapshot snapshot,
+                                                 SnapshotSource snapshotSource) {
+        // TODO this will need to be changed when we have more than one dataset per snapshot (>1 contentsModel)
+        List<SnapshotTable> tableList = new ArrayList<>();
+        snapshot.snapshotTables(tableList);
+        List<SnapshotMapTable> mapTableList = new ArrayList<>();
+        snapshotSource.snapshotMapTables(mapTableList);
+        Dataset dataset = snapshotSource.getDataset();
+
+        // for each dataset table specified in the request, create a table in the snapshot with the same name
+        for (DatasetTable datasetTable : dataset.getTables()) {
+            List<Column> columnList = new ArrayList<>();
+            SnapshotTable snapshotTable = new SnapshotTable()
+                .name(datasetTable.getName())
+                .columns(columnList);
+            tableList.add(snapshotTable);
+            List<SnapshotMapColumn> mapColumnList = new ArrayList<>();
+            mapTableList.add(new SnapshotMapTable()
+                .fromTable(datasetTable)
+                .toTable(snapshotTable)
+                .snapshotMapColumns(mapColumnList));
+
+            // for each dataset column specified in the request, create a column in the snapshot with the same name
+            datasetTable.getColumns()
```

Comment:
This is not incorrect, but the ordering seems odd to me.
It would make more sense to my brain if you built the mapColumnList and the columnList first and then filled the result into the snapshotTable.